In [ ]:
## read column names
# print(nr.columns)

## read each column
# print(nr['food_type'])

## read each row
# print(nr.iloc[0:4])
# nr.loc[nr['food_type']=="CHEESE"]

## read a specific location (x,y)
# print(nr.iloc[2,1])

In [36]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [37]:
csv_path = "nutrient_retention_clean.csv"

In [38]:
nr=pd.read_csv(csv_path)

In [39]:
nr

retn_code  foodgrp_code food_type    food_desc cook_method_1  \
0             1             1    CHEESE          NaN         BAKED   
1             1             1    CHEESE          NaN         BAKED   
2             1             1    CHEESE          NaN         BAKED   
3             1             1    CHEESE          NaN         BAKED   
4             1             1    CHEESE          NaN         BAKED   
...         ...           ...       ...          ...           ...   
6961        433            20      RICE  WHITE/BROWN       SAUTEED   
6962        433            20      RICE  WHITE/BROWN       SAUTEED   
6963        433            20      RICE  WHITE/BROWN       SAUTEED   
6964        433            20      RICE  WHITE/BROWN       SAUTEED   
6965        433            20      RICE  WHITE/BROWN       SAUTEED   

     cook_method_2 cook_method_addon cook_time  nutrient_num  \
0              NaN               NaN       NaN           301   
1              NaN               NaN       NaN           303   
2              NaN               NaN       NaN           304   
3              NaN               NaN       NaN           305   
4              NaN               NaN       NaN           306   
...            ...               ...       ...           ...   
6961      SIMMERED               NaN       NaN           321   
6962      SIMMERED               NaN       NaN           322   
6963      SIMMERED               NaN       NaN           334   
6964      SIMMERED               NaN       NaN           337   
6965      SIMMERED               NaN       NaN           338   

            nutrient_desc  retn_factor  
0                 Calcium        100.0  
1                    Iron        100.0  
2               Magnesium        100.0  
3              Phosphorus        100.0  
4               Potassium        100.0  
...                   ...          ...  
6961       Carotene, beta         90.0  
6962      Carotene, alpha         90.0  
6963  Cryptoxanthin, beta         90.0  
6964             Lycopene         90.0  
6965  Lutein + zeaxanthin         90.0  

[6966 rows x 11 columns]

In [40]:
ca=nr.loc[nr['nutrient_desc']=='Calcium']
ca.head(7)

retn_code  foodgrp_code food_type food_desc cook_method_1 cook_method_2  \
0            1             1    CHEESE       NaN         BAKED           NaN   
26           3             1    CHEESE       NaN       BROILED           NaN   
52           5             1    CHEESE       NaN        COOKED           NaN   
78           7             1    CHEESE       NaN      REHEATED           NaN   
104        101             1      EGGS       NaN         BAKED           NaN   
130        103             1      EGGS       NaN         FRIED     SCRAMBLED   
156        105             1      EGGS       NaN   HARD COOKED           NaN   

    cook_method_addon cook_time  nutrient_num nutrient_desc  retn_factor  
0                 NaN       NaN           301       Calcium        100.0  
26                NaN       NaN           301       Calcium        100.0  
52           W/LIQUID       NaN           301       Calcium        100.0  
78                NaN       NaN           301       Calcium        100.0  
104               NaN       NaN           301       Calcium        100.0  
130               NaN       NaN           301       Calcium        100.0  
156               NaN       NaN           301       Calcium        100.0

In [41]:
eggs=nr.loc[nr['food_type']=='EGGS']
eggs.head(7)

retn_code  foodgrp_code food_type food_desc cook_method_1 cook_method_2  \
104        101             1      EGGS       NaN         BAKED           NaN   
105        101             1      EGGS       NaN         BAKED           NaN   
106        101             1      EGGS       NaN         BAKED           NaN   
107        101             1      EGGS       NaN         BAKED           NaN   
108        101             1      EGGS       NaN         BAKED           NaN   
109        101             1      EGGS       NaN         BAKED           NaN   
110        101             1      EGGS       NaN         BAKED           NaN   

    cook_method_addon cook_time  nutrient_num nutrient_desc  retn_factor  
104               NaN       NaN           301       Calcium        100.0  
105               NaN       NaN           303          Iron        100.0  
106               NaN       NaN           304     Magnesium        100.0  
107               NaN       NaN           305    Phosphorus        100.0  
108               NaN       NaN           306     Potassium        100.0  
109               NaN       NaN           307        Sodium        100.0  
110               NaN       NaN           309          Zinc        100.0

In [50]:
nr[nr['retn_factor']==100.0]

retn_code  foodgrp_code food_type    food_desc cook_method_1  \
0             1             1    CHEESE          NaN         BAKED   
1             1             1    CHEESE          NaN         BAKED   
2             1             1    CHEESE          NaN         BAKED   
3             1             1    CHEESE          NaN         BAKED   
4             1             1    CHEESE          NaN         BAKED   
...         ...           ...       ...          ...           ...   
6946        433            20      RICE  WHITE/BROWN       SAUTEED   
6951        433            20      RICE  WHITE/BROWN       SAUTEED   
6956        433            20      RICE  WHITE/BROWN       SAUTEED   
6957        433            20      RICE  WHITE/BROWN       SAUTEED   
6960        433            20      RICE  WHITE/BROWN       SAUTEED   

     cook_method_2 cook_method_addon cook_time  nutrient_num   nutrient_desc  \
0              NaN               NaN       NaN           301         Calcium   
1              NaN               NaN       NaN           303            Iron   
2              NaN               NaN       NaN           304       Magnesium   
3              NaN               NaN       NaN           305      Phosphorus   
4              NaN               NaN       NaN           306       Potassium   
...            ...               ...       ...           ...             ...   
6946      SIMMERED               NaN       NaN           309            Zinc   
6951      SIMMERED               NaN       NaN           406          Niacin   
6956      SIMMERED               NaN       NaN           421  Choline, total   
6957      SIMMERED               NaN       NaN           418    Vitamin B-12   
6960      SIMMERED               NaN       NaN           221  Alcohol, ethyl   

      retn_factor  
0           100.0  
1           100.0  
2           100.0  
3           100.0  
4           100.0  
...           ...  
6946        100.0  
6951        100.0  
6956        100.0  
6957        100.0  
6960        100.0  

[2601 rows x 11 columns]

In [58]:
frozen=nr[nr['food_desc']=='FROZEN']
frozen
# frozen[frozen['retn_factor']>=90.0]

retn_code  foodgrp_code food_type food_desc cook_method_1 cook_method_2  \
1092        156             9    FRUITS    FROZEN           NaN           NaN   
1093        156             9    FRUITS    FROZEN           NaN           NaN   
1094        156             9    FRUITS    FROZEN           NaN           NaN   
1095        156             9    FRUITS    FROZEN           NaN           NaN   
1096        156             9    FRUITS    FROZEN           NaN           NaN   
...         ...           ...       ...       ...           ...           ...   
4518       2605            15   FINFISH    FROZEN        HEATED           NaN   
4519       2605            15   FINFISH    FROZEN        HEATED           NaN   
4520       2605            15   FINFISH    FROZEN        HEATED           NaN   
4521       2605            15   FINFISH    FROZEN        HEATED           NaN   
4522       2605            15   FINFISH    FROZEN        HEATED           NaN   

     cook_method_addon cook_time  nutrient_num        nutrient_desc  \
1092               NaN       NaN           301              Calcium   
1093               NaN       NaN           303                 Iron   
1094               NaN       NaN           304            Magnesium   
1095               NaN       NaN           305           Phosphorus   
1096               NaN       NaN           306            Potassium   
...                ...       ...           ...                  ...   
4518               NaN       NaN           321       Carotene, beta   
4519               NaN       NaN           322      Carotene, alpha   
4520               NaN       NaN           334  Cryptoxanthin, beta   
4521               NaN       NaN           337             Lycopene   
4522               NaN       NaN           338  Lutein + zeaxanthin   

      retn_factor  
1092         95.0  
1093        100.0  
1094        100.0  
1095        100.0  
1096         90.0  
...           ...  
4518         90.0  
4519         90.0  
4520         90.0  
4521         90.0  
4522         90.0  

[234 rows x 11 columns]

In [85]:
nr['food_type'].drop_duplicates()

0                      CHEESE
104                      EGGS
234                      MILK
338                   CHICKEN
572                    TURKEY
676                    CEREAL
728                   OATMEAL
780                    FRUITS
910            FRUITS(CITRUS)
962        FRUITS(NOT CITRUS)
1118                    BACON
1144              HAM(PICNIC)
1196                     PORK
1820            SAUSAGE(PORK)
1924                  SAUSAGE
1950          SAUSAGE(FRANKS)
2054                 POTATOES
2470            SWEETPOTATOES
2704                 TOMATOES
2782               VEG/GREENS
2964                VEG/OTHER
3198                VEG/ROOTS
3432                     NUTS
3536                     BEEF
4056                    LIVER
4108    ORGAN MEATS(NOTLIVER)
4212                  ALC BEV
4471                  FINFISH
4965      SEAFOOD/FROG/TURTLE
5251                SHELLFISH
5511                  LEGUMES
6005                     LAMB
6264                     VEAL
6602      

# Notes on Nutrient Retention factors
### <70
641 rows
### >=70
6301 rows
### =100
2601 rows

## Frozen (total = 234 rows)
### >= 90
199 rows

In [59]:
grouped = nr.groupby(by="cook_method_1")
grouped

In [60]:
type(grouped)

pandas.core.groupby.generic.DataFrameGroupBy

In [61]:
grouped.groups

{'BAKED': [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 962, 963, 964, 965, 966, 967, 968, 969, 970, 971, 972, 973, 974, 975, 976, 977, 978, 979, 980, 981, 982, 983, 984, 985, 986, 987, 2080, 2081, 2082, 2083, 2084, 2085, 2086, 2087, 2088, 2089, 2090, 2091, 2092, 2093, 2094, 2095, 2096, 2097, 2098, 2099, 2100, 2101, ...], 'BAKED IN CASSEROLE': [2054, 2055, 2056, 2057, 2058, 2059, 2060, 2061, 2062, 2063, 2064, 2065, 2066, 2067, 2068, 2069, 2070, 2071, 2072, 2073, 2074, 2075, 2076, 2077, 2078, 2079, 2496, 2497, 2498, 2499, 2500, 2501, 2502, 2503, 2504, 2505, 2506, 2507, 2508, 2509, 2510, 2511, 2512, 2513, 2514, 2515, 2516, 2517, 2518, 2519, 2520, 2521], 'BOILED': [2106, 2107, 2108, 2109, 2110, 2111, 2112, 2113, 2114, 2115, 2116, 2117, 2118, 2119, 2120, 2121, 2122, 2123, 2124, 2125, 2126, 2127, 2128, 2129, 2130, 

In [66]:
grouped['retn_factor'].median().sort_values()

cook_method_1
BROWNED                75.0
BRAISED                80.0
SIMMERED               80.0
ROASTED                80.0
BOILED                 90.0
BROILED                90.0
DRIED                  90.0
STEWED                 90.0
FRIED                  90.0
STEAMED                90.0
SAUTEED                90.0
STIR FRY               95.0
COOKED                 95.0
MASHED                 95.0
BAKED                 100.0
HEATED                100.0
POACHED               100.0
HOT LIQUID            100.0
STORED OVERNIGHT      100.0
HARD COOKED           100.0
FLAMED                100.0
BAKED IN CASSEROLE    100.0
REHEATED              100.0
TOASTED               100.0
Name: retn_factor, dtype: float64

In [67]:
grouped['retn_factor'].mean().sort_values()

cook_method_1
BROWNED               78.689904
SIMMERED              81.422065
DRIED                 81.730769
ROASTED               82.115385
BRAISED               82.532051
STEWED                83.461538
BOILED                84.842657
SAUTEED               85.439560
BROILED               87.939560
FRIED                 88.575499
COOKED                90.985577
STEAMED               91.807692
POACHED               92.307692
HEATED                92.500000
MASHED                93.076923
BAKED                 93.456615
STIR FRY              93.461538
BAKED IN CASSEROLE    93.653846
HARD COOKED           93.846154
TOASTED               95.576923
REHEATED              98.624484
STORED OVERNIGHT      98.846154
FLAMED                99.038462
HOT LIQUID            99.423077
Name: retn_factor, dtype: float64

In [68]:
grouped2 = nr.groupby(by="cook_method_2")
grouped2

In [69]:
grouped2['retn_factor'].median().sort_values()

cook_method_2
FRIED            80.0
SIMMERED         80.0
BAKED            85.0
LITTLE WATER     90.0
WATER COVER      90.0
BROILED          95.0
BROWNED          95.0
SCRAMBLED       100.0
STUFFED         100.0
Name: retn_factor, dtype: float64

In [70]:
grouped2['retn_factor'].mean().sort_values()

cook_method_2
FRIED           75.544872
BAKED           79.252137
SIMMERED        83.136000
WATER COVER     87.948718
BROWNED         88.750000
LITTLE WATER    89.615385
BROILED         92.500000
SCRAMBLED       94.038462
STUFFED         95.384615
Name: retn_factor, dtype: float64

In [80]:
grouped3 = nr.groupby(by=["cook_method_1","cook_method_2"])
grouped3["retn_factor"].describe().sort_values(by="mean")

count       mean        std   min    25%    50%  \
cook_method_1 cook_method_2                                                    
BOILED        FRIED          156.0  75.544872  18.055235  25.0   65.0   80.0   
              BAKED          208.0  78.149038  18.430229  25.0   65.0   85.0   
BROWNED       SIMMERED       416.0  78.689904  15.732018  30.0   70.0   75.0   
ROASTED       SIMMERED        52.0  80.480769  15.251474  40.0   70.0   80.0   
BOILED        WATER COVER     78.0  87.948718  11.493530  55.0   85.0   90.0   
DRIED         BAKED           26.0  88.076923  12.415871  60.0   85.0   90.0   
BOILED        BROWNED         52.0  88.750000  16.799714  25.0   90.0   95.0   
SAUTEED       SIMMERED        26.0  88.846154  12.985199  60.0   90.0   90.0   
BOILED        LITTLE WATER    26.0  89.615385   9.265320  70.0   90.0   90.0   
FRIED         BROILED         26.0  92.500000   9.300538  70.0   90.0   95.0   
              SCRAMBLED       26.0  94.038462   9.167249  75.0   87.5  100.0   
BAKED         STUFFED         26.0  95.384615   7.864233  80.0   95.0  100.0   
              SIMMERED       131.0  97.175573  14.386148  10.0  100.0  100.0   

                               75%    max  
cook_method_1 cook_method_2                
BOILED        FRIED           85.0  100.0  
              BAKED           90.0  100.0  
BROWNED       SIMMERED        95.0  100.0  
ROASTED       SIMMERED       100.0  100.0  
BOILED        WATER COVER     95.0  100.0  
DRIED         BAKED          100.0  100.0  
BOILED        BROWNED        100.0  100.0  
SAUTEED       SIMMERED       100.0  100.0  
BOILED        LITTLE WATER    95.0  100.0  
FRIED         BROILED        100.0  100.0  
              SCRAMBLED      100.0  100.0  
BAKED         STUFFED        100.0  100.0  
              SIMMERED       100.0  100.0

In [82]:
grouped["retn_factor"].describe().sort_values(by="mean")

count       mean        std   min     25%    50%    75%  \
cook_method_1                                                                  
BROWNED              416.0  78.689904  15.732018  30.0   70.00   75.0   95.0   
SIMMERED             988.0  81.422065  15.022003  30.0   75.00   80.0   95.0   
DRIED                 52.0  81.730769  21.141090  20.0   60.00   90.0  100.0   
ROASTED              260.0  82.115385  12.417533  40.0   75.00   80.0   95.0   
BRAISED              156.0  82.532051  14.429996  35.0   75.00   80.0   95.0   
STEWED                52.0  83.461538  16.073924  50.0   75.00   90.0  100.0   
BOILED              1144.0  84.842657  15.865844  25.0   80.00   90.0   95.0   
SAUTEED              182.0  85.439560  15.080783  30.0   75.00   90.0  100.0   
BROILED              546.0  87.939560  11.660881  30.0   80.00   90.0  100.0   
FRIED                702.0  88.575499  10.899325  45.0   80.00   90.0  100.0   
COOKED               416.0  90.985577  12.553108  30.0   90.00   95.0  100.0   
STEAMED              130.0  91.807692   8.117448  70.0   90.00   90.0  100.0   
POACHED               26.0  92.307692  10.220643  75.0   81.25  100.0  100.0   
HEATED               104.0  92.500000  13.563944  30.0   90.00  100.0  100.0   
MASHED                26.0  93.076923   8.952868  75.0   95.00   95.0  100.0   
BAKED                703.0  93.456615  11.303910  10.0   90.00  100.0  100.0   
STIR FRY              78.0  93.461538   6.561779  80.0   90.00   95.0  100.0   
BAKED IN CASSEROLE    52.0  93.653846   8.972473  75.0   90.00  100.0  100.0   
HARD COOKED           26.0  93.846154   9.413574  75.0   87.50  100.0  100.0   
TOASTED               26.0  95.576923   6.217222  80.0   95.00  100.0  100.0   
REHEATED             727.0  98.624484   2.783383  70.0  100.00  100.0  100.0   
STORED OVERNIGHT      26.0  98.846154   5.883484  70.0  100.00  100.0  100.0   
FLAMED                26.0  99.038462   4.902903  75.0  100.00  100.0  100.0   
HOT LIQUID            26.0  99.423077   2.941742  85.0  100.00  100.0  100.0   

                      max  
cook_method_1              
BROWNED             100.0  
SIMMERED            100.0  
DRIED               100.0  
ROASTED             100.0  
BRAISED             100.0  
STEWED              100.0  
BOILED              100.0  
SAUTEED             100.0  
BROILED             100.0  
FRIED               100.0  
COOKED              100.0  
STEAMED             100.0  
POACHED             100.0  
HEATED              100.0  
MASHED              100.0  
BAKED               100.0  
STIR FRY            100.0  
BAKED IN CASSEROLE  100.0  
HARD COOKED         100.0  
TOASTED             100.0  
REHEATED            100.0  
STORED OVERNIGHT    100.0  
FLAMED              100.0  
HOT LIQUID          100.0

In [90]:
rootveg=nr.loc[nr['food_type']=='VEG/ROOTS']
rootveg[rootveg['cook_method_1']=="BAKED"]

retn_code  foodgrp_code  food_type food_desc cook_method_1  \
3198       3451            11  VEG/ROOTS       NaN         BAKED   
3199       3451            11  VEG/ROOTS       NaN         BAKED   
3200       3451            11  VEG/ROOTS       NaN         BAKED   
3201       3451            11  VEG/ROOTS       NaN         BAKED   
3202       3451            11  VEG/ROOTS       NaN         BAKED   
3203       3451            11  VEG/ROOTS       NaN         BAKED   
3204       3451            11  VEG/ROOTS       NaN         BAKED   
3205       3451            11  VEG/ROOTS       NaN         BAKED   
3206       3451            11  VEG/ROOTS       NaN         BAKED   
3207       3451            11  VEG/ROOTS       NaN         BAKED   
3208       3451            11  VEG/ROOTS       NaN         BAKED   
3209       3451            11  VEG/ROOTS       NaN         BAKED   
3210       3451            11  VEG/ROOTS       NaN         BAKED   
3211       3451            11  VEG/ROOTS       NaN         BAKED   
3212       3451            11  VEG/ROOTS       NaN         BAKED   
3213       3451            11  VEG/ROOTS       NaN         BAKED   
3214       3451            11  VEG/ROOTS       NaN         BAKED   
3215       3451            11  VEG/ROOTS       NaN         BAKED   
3216       3451            11  VEG/ROOTS       NaN         BAKED   
3217       3451            11  VEG/ROOTS       NaN         BAKED   
3218       3451            11  VEG/ROOTS       NaN         BAKED   
3219       3451            11  VEG/ROOTS       NaN         BAKED   
3220       3451            11  VEG/ROOTS       NaN         BAKED   
3221       3451            11  VEG/ROOTS       NaN         BAKED   
3222       3451            11  VEG/ROOTS       NaN         BAKED   
3223       3451            11  VEG/ROOTS       NaN         BAKED   

     cook_method_2 cook_method_addon cook_time  nutrient_num  \
3198           NaN               NaN       NaN           301   
3199           NaN               NaN       NaN           303   
3200           NaN               NaN       NaN           304   
3201           NaN               NaN       NaN           305   
3202           NaN               NaN       NaN           306   
3203           NaN               NaN       NaN           307   
3204           NaN               NaN       NaN           309   
3205           NaN               NaN       NaN           312   
3206           NaN               NaN       NaN           401   
3207           NaN               NaN       NaN           404   
3208           NaN               NaN       NaN           405   
3209           NaN               NaN       NaN           406   
3210           NaN               NaN       NaN           415   
3211           NaN               NaN       NaN           417   
3212           NaN               NaN       NaN           431   
3213           NaN               NaN       NaN           432   
3214           NaN               NaN       NaN           421   
3215           NaN               NaN       NaN           418   
3216           NaN               NaN       NaN           318   
3217           NaN               NaN       NaN           392   
3218           NaN               NaN       NaN           221   
3219           NaN               NaN       NaN           321   
3220           NaN               NaN       NaN           322   
3221           NaN               NaN       NaN           334   
3222           NaN               NaN       NaN           337   
3223           NaN               NaN       NaN           338   

                       nutrient_desc  retn_factor  
3198                         Calcium        100.0  
3199                            Iron        100.0  
3200                       Magnesium        100.0  
3201                      Phosphorus        100.0  
3202                       Potassium        100.0  
3203                          Sodium        100.0  
3204                            Zinc        100.0  
3205                    